Libraries/src/Amazon.Lambda.CognitoEvents/CognitoPreTokenGenerationRequest.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace Amazon.Lambda.CognitoEvents
+{
+    /// <summary>
+    /// https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html
+    /// </summary>
+    public class CognitoPreTokenGenerationRequest : CognitoTriggerRequest
+    {
+        /// <summary>
+        /// The input object containing the current group configuration. It includes groupsToOverride, iamRolesToOverride, and preferredRole.
+        /// </summary>
+        [DataMember(Name = "groupConfiguration")]
```

Comment:
These should be `JsonPropertyName` not `DataMember`

Default behaviour of the serializer is to use PascalCase, and Cognito uses camelCase, so these are essential

Reply:
`JsonPropertyName` is included for NetCoreApp3.1.  I followed the patterns called out in examples such as 

https://github.com/aws/aws-lambda-dotnet/blob/087590ce99274e16e26d37e1dfd73b0b71d1230a/Libraries/src/Amazon.Lambda.ApplicationLoadBalancerEvents/ApplicationLoadBalancerResponse.cs

with included tests to verify serialization for all supported targets, and under all situations we're getting the correct camelCase serialization.  Is this pattern for calling out `DataMember` and `JsonPropertyName` deprecated and something that should now be avoided in this library?

Reply:
I've used your code in .Net 6 and it fails. Adding `JsonPropertyName` fixes it.

Reply:
I'll take another look at it.  I suspect the majority of serialization classes within this repo would exhibit this issue, given this is an existing pattern.

Reply:
I think it's as simple as removing the conditional and always having the JsonPropertyName attribute present